src/main/java/repo/impl/PostgresSessionRepo.java

```
@@ -0,0 +1,129 @@
+ */
+@Repository
+public class PostgresSessionRepo implements SessionRepo{
+
+    @Autowired
+    @Qualifier("jdbcFormplayer")
+    private JdbcTemplate jdbcTemplate;
+
+    @Override
+    public void save(SerializableFormSession session) {
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        ObjectOutputStream oos;
+        try {
+            oos = new ObjectOutputStream(baos);
+            oos.writeObject(session.getSessionData());
+            byte[] sessionDataBytes = baos.toByteArray();
+
+            int sessionCount = this.jdbcTemplate.queryForObject(
+                    replaceTableName("select count(*) from %s where id = ?"), Integer.class, session.getId());
+
+            if(sessionCount > 0){
+                String query = replaceTableName("UPDATE %s SET instanceXml = ?, sessionData = ? WHERE id = ?");
+                this.jdbcTemplate.update(query,  new Object[] {session.getInstanceXml(), sessionDataBytes, session.getId()},
+                        new int[] {Types.VARCHAR, Types.BINARY, Types.VARCHAR});
+                return;
+            }
+
+            String query = replaceTableName("INSERT into %s " +
+                    "(id, instanceXml, formXml, " +
+                    "restoreXml, username, initLang, sequenceId, " +
+                    "domain, postUrl, sessionData) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)");
+            this.jdbcTemplate.update(query,  new Object[] {session.getId(), session.getInstanceXml(), session.getFormXml(),
+                    session.getRestoreXml(), session.getUsername(), session.getInitLang(), session.getSequenceId(),
+                    session.getDomain(), session.getPostUrl(), sessionDataBytes}, new int[] {
+                    Types.VARCHAR, Types.VARCHAR, Types.VARCHAR, Types.VARCHAR, Types.VARCHAR,
+                    Types.VARCHAR, Types.VARCHAR, Types.VARCHAR, Types.VARCHAR, Types.BINARY});
+        } catch (IOException e) {
+            e.printStackTrace();
```

or at least fail hard instead of swallowing and continuing as if we successfully saved